salt/_states/metalk8s_cordon.py

```diff
@@ -70,4 +69,25 @@ def node_uncordoned(name, **kwargs):
     name
         the name of the node
     '''
-    return _node_set_unschedulable(name, False, **kwargs)
+    # k-c-m 1.12 adds taint `node.kubernetes.io/unschedulable` when cordonning
+    # a node but k-c-m 1.11 does not remove this taint when uncordonning so,
+    # we need to manually remove it when necessary by patching taints.
+
+    all_node_taints = __salt__['metalk8s_kubernetes.node_taints'](
+        name, **kwargs
+    )
+    key_to_remove = "node.kubernetes.io/unschedulable"
+
+    new_taints = [
+        taint for taint in all_node_taints
+        if not (taint['key'] == key_to_remove)
+    ]
+    if new_taints == all_node_taints:
```

Comment:
That's going to be flaky, I don't think we should trust ordering in the API response.

You have multiple solutions:

- You could make the taint `dict`s hashable (e.g. `serialized = "key={};value={};effect={}".format(t['key'], t['value'], t['effect'])`), then compare `set`s of these serialized values
- A **much** simpler/efficient approach is to test `if key_to_remove in (t['key'] for t in all_node_taints):` before attempting to remove the taint

Reply:
I think the order should be the same as we create the `new_taints` list using `all_node_taints`.

But I agree with the second point check this before "computing" the `new_taints` list it's much better